### PR TITLE
Core/Opcodes: Implement CMSG_GROUP_SWAP_SUB_GROUP

### PR DESCRIPTION
**Pull request by FALL1N1**
## CHANGES PROPOSED:
-  Implement the group swap opcode 

## ISSUES ADDRESSED:
- none

## TESTS PERFORMED:
- Read below

## HOW TO TEST THE CHANGES:
Log with two characters, make a raid group, place the 2nd char in a different group, then drag your char's bar to the other char's bar. 
Without this commit = nothing happens, with it = you will swap your group with the other char.
Enjoy your new server feature.

## KNOWN ISSUES AND TODO LIST:

- None

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/game/Handlers/GroupHandler.cpp [hunk 1]
```diff
@@ -1134,3 +1134,69 @@ void WorldSession::HandleOptOutOfLootOpcode(WorldPacket& recvData)
...
+    Group* group = GetPlayer()->GetGroup();
+    if (!group || !group->isRaidGroup())
+        return;
+
+    if (!group->IsLeader(GetPlayer()->GetGUID()) &&
+        !group->IsAssistant(GetPlayer()->GetGUID()))
+        return;
+
+    //get guid, member may be offline
+    auto getGuid = [&group](std::string const& playerName)
```
Comment (FrancescoBorzi) [inline src/server/game/Handlers/GroupHandler.cpp:1165]: nice that you use `auto`, consider using it also in other places ;)

File: src/server/game/Handlers/GroupHandler.cpp [hunk 2]
```diff
@@ -1134,3 +1134,69 @@ void WorldSession::HandleOptOutOfLootOpcode(WorldPacket& recvData)
...
+    {
+        // no player, cheating?
+        if (!group->GetMemberGUID(playerName))
+            return uint64(0);
+
+        if (Player* player = ObjectAccessor::FindPlayerByName(playerName.c_str()))
+            return player->GetGUID();
```
Comment (FrancescoBorzi) [inline src/server/game/Handlers/GroupHandler.cpp:1172]: brackets please

File: src/server/game/Handlers/GroupHandler.cpp (hunk 1 above)
Comment (FALL1N1) [reply src/server/game/Handlers/GroupHandler.cpp:1165]: I just back port the code as-is (as long as it works and it is an improvement), if you find something that can be optimized just leave a comment just like now and I will rewrite it.

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by pklloveyou** [comment]
Tested

**Comment by Kitzunu** [comment]
Thanks!

**Pull request merged**